botocore/endpoint.py

```diff
@@ -231,6 +231,7 @@ def _get_response(self, request, operation_model, attempts):
             operation_model.metadata['protocol'])
         parsed_response = parser.parse(
             response_dict, operation_model.output_shape)
+        http_response.close()
```

Comment:
I think this would be better written as:

```python
with http_response:
    response_dict = ...
    parser = ...
    parsed_response = ...
    return (http_response, parsed_response), None
```

It has the same effect, but it ensures it gets closed even if there is a exception here.